pkg/xds/envoy/routes/v3/routes_configurer.go

```diff
@@ -163,7 +163,8 @@ func (c RoutesConfigurer) hasExternal(clusters []envoy_common.Cluster) bool {
 func (c RoutesConfigurer) routeAction(clusters []envoy_common.Cluster, modify *mesh_proto.TrafficRoute_Http_Modify) *envoy_route.RouteAction {
 	routeAction := &envoy_route.RouteAction{}
 	if len(clusters) != 0 {
-		routeAction.Timeout = util_proto.Duration(clusters[0].Timeout().GetHttp().GetRequestTimeout().AsDuration())
+		cluster := clusters[0].(*envoy_common.ClusterImpl)
```

Comment:
why is it ok to look at only the first one? maybe a comment? or maybe it indicates the timeout shouldn't be on the split? :thinking:

Reply:
added a comment